python/lsst/pipe/base/task.py

```diff
@@ -111,7 +111,7 @@ def __init__(self, config=None, name=None, parentTask=None, log=None):
             if config is None:
                 config = getattr(parentTask.config, name)
             self._taskDict = parentTask._taskDict
-            self.log = Log.getLogger(parentTask.log.getName() + '.' + name)
+            loggerName = parentTask.log.getName() + '.' + name
```

Comment:
Your change agreed with my comment to get rid of duplication of all the `Log.getLogger` calls and allow the logic to focus on the real issue here which was choosing the name to pass into `getLogger`. I don't see how your change changes the API at all.